recipe/meta.yaml

```
@@ -1,61 +1,204 @@
+    folder: streamlit
+  - url: https://github.com/{{ name }}/{{ name }}/archive/{{ version }}.tar.gz
+    sha256: d13d8ed5f0b8b429febe0868619d7b42298e7badba266ee7fa73823039ce7de8
+    folder: streamlit_tests
 
 build:
   number: 0
-  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
-  entry_points:
-    - streamlit = streamlit.web.cli:main
-  skip: true  # [s390x or py<38]
+  # toml is not avaaliable on linux-s390x
+  skip: True  # [py<39]
+  # https://github.com/streamlit/streamlit/blob/1.42.0/lib/setup.py#L143
+  # python_requires=">=3.9, !=3.9.7"
+  skip: True  # [s390x]
+  # Watchdog is not avaliable on win + py313
+  skip: True  # [win and py>312]
 
 requirements:
   host:
-    - pip
     - python
-    - wheel
-    - setuptools
   run:
     - python
-    - altair >=4.0,<6
-    - blinker >=1.0.0,<2
-    - cachetools >=4.0,<6
-    - click >=7.0,<9
-    - numpy >=1.23,<3
-    - packaging >=20,<25
-    - pandas >=1.4.0,<3
-    - pillow >=7.1.0,<12
-    - protobuf >=3.20,<6
-    - pyarrow >=7.0
-    - requests >=2.27,<3
-    - rich >=10.14.0,<14
-    - tenacity >=8.1.0,<10
-    - toml >=0.10.1,<2
-    - typing_extensions >=4.3.0,<5
-    - watchdog >=2.1.5,<7  # [not osx]
-    - gitpython >=3.0.7,<4,!=3.1.19
-    - pydeck >=0.8.0b4,<1
-    - tornado >=6.0.3,<7
-  run_constrained:
-    - snowflake-snowpark-python >=1.17.0  # [py<312]
-    - snowflake-connector-python >=2.8.0  # [py<312]
-    - python !=3.9.7
-
-test:
-  imports:
-    - streamlit
-  requires:
-    - pip
-  commands:
-    - pip check
-    - streamlit --help
+
+outputs:
+  - name: streamlit
+    script: build-output.sh  # [not win]
+    script: build-output.bat  # [win]
+    build:
+      entry_points:
+        - streamlit = streamlit.web.cli:main
+    requirements:
+      host:
+        - pip
+        - python
+        - wheel
+        - setuptools
+      run:
+        - python
+        - altair >=4.0,<6
+        - blinker >=1.0.0,<2
+        - cachetools >=4.0,<6
+        - click >=7.0,<9
+        - numpy >=1.23,<3
+        - packaging >=20,<25
+        - pandas >=1.4.0,<3
+        - pillow >=7.1.0,<12
+        - protobuf >=3.20,<6
+        - pyarrow >=7.0
+        - requests >=2.27,<3
+        - rich >=10.14.0,<14
+        - tenacity >=8.1.0,<10
+        - toml >=0.10.1,<2
+        - typing_extensions >=4.4.0,<5
+        - watchdog >=2.1.5,<7  # [not osx]
+        - gitpython >=3.0.7,<4,!=3.1.19
+        - pydeck >=0.8.0b4,<1
+        - tornado >=6.0.3,<7
+      run_constrained:
+        - snowflake-snowpark-python >=1.17.0  # [py<312]
+        - snowflake-connector-python >=3.3.0  # [py<312]
+        - python !=3.9.7
+    test:
+      imports:
+        - streamlit
+      commands:
+        - pip check
+        - streamlit --help
+      requires:
+        - pip
+
+  - name: streamlit-tests
+    build:
+    requirements:
+      host:
+        - {{ pin_subpackage('streamlit', exact=True) }}
+      run:
+        - {{ pin_subpackage('streamlit', exact=True) }}
```

```suggestion
      host:
      	- python
     run:
        - python
        - {{ pin_subpackage('streamlit', exact=True) }}
```
Without python in host and run, it produces only one variant of streamlit-tests and pin to a random output of streamlit.